inspirehep/modules/records/serializers/schemas/json.py

```diff
@@ -36,3 +41,98 @@ class RecordSchemaJSONUIV1(Schema):
     links = fields.Raw()
     created = fields.Str()
     updated = fields.Str()
+
+
+class MetadataReferencesSchemaItemV1(Schema):
+    arxiv_eprints = fields.List(fields.Dict())
+    authors = fields.Method('get_authors')
+    collaborations = fields.List(fields.Dict())
+    control_number = fields.Int()
+    dois = fields.List(fields.Dict())
+    publication_info = fields.List(fields.Dict())
+    titles = fields.List(fields.Dict())
+
+    def get_authors(self, data):
+        authors = get_value(data, 'authors', default=[])
+        return authors[:10] or missing
+
+
+class MetadataReferencesSchemaUIV1(Schema):
+    references = fields.Nested(
+        MetadataReferencesSchemaItemV1, many=True, dump_only=True)
+
+
+class ReferencesSchemaJSONUIV1(RecordSchemaJSONUIV1):
+    """Schema for references."""
+    metadata = fields.Nested(MetadataReferencesSchemaUIV1, dump_only=True)
+
+    @pre_dump
+    def pre_process(self, data):
+        resolved = []
+        references = data['metadata'].get('references')
+
+        if references is not None:
+            reference_record_ids = [
+                get_recid_from_ref(reference.get('record'))
+                for reference in references if reference.get('record')
+            ]
+            resolved_references = get_resolved_references(reference_record_ids)
+            recid_to_reference = {
+                reference['control_number']: reference
+                for reference in resolved_references
+            }
+
+            for reference in references:
+                if 'record' in reference:
+                    reference_record_id = get_recid_from_ref(
+                        reference.get('record'))
+                    reference_record = recid_to_reference.get(
+                        reference_record_id)
+
+                    if reference_record:
+                        resolved.append(reference_record)
+                        break
+
+                if 'reference' in reference:
```

Comment:
I think this would be clearer if you did this part in a more declarative style. Currently it's quite hard to follow.
Something like
```python
publication_info = force_list(
    reference['publication_info'] if {'journal_title', 'pubinfo_freetext'}.issubset(reference.get('publication_info', {}).keys()) else None
)
# etc.
reference.update({
    'publication_info': publication_info,
    # etc.
})
del reference['reference']
strip_empty_values(reference)
```